client/lib/domains/assembler.js

```diff
@@ -23,6 +23,7 @@ function createDomainObjects( dataTransferObject ) {
 		return {
 			autoRenewalMoment: domain.auto_renewal_date && i18n.moment( domain.auto_renewal_date ),
 			currentUserCanManage: domain.current_user_can_manage,
+			domainLockingAvailable: domain.domain_locking_available,
```

Comment:
Could you also add this to https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/domains/assembler.js?